File: mobile/android/src/main/kotlin/com/otravelz/android/data/network/dto/WeatherDtos.kt
package com.otravelz.android.data.network.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CurrentWeatherDto(
    @SerialName("location_name") val locationName: String? = null,
    val lat: Double? = null,
    val lon: Double? = null,
    @SerialName("observed_at") val observedAt: String? = null,
    @SerialName("temperature_c") val temperatureC: Double? = null,
    @SerialName("apparent_temperature_c") val apparentTemperatureC: Double? = null,
    val condition: String? = null,
    @SerialName("condition_code") val conditionCode: Int? = null,
    @SerialName("is_day") val isDay: Int? = null,
    @SerialName("humidity_pct") val humidityPct: Int? = null,
    @SerialName("precipitation_probability_pct") val precipitationProbabilityPct: Int? = null,
    @SerialName("precipitation_mm") val precipitationMm: Double? = null,
    @SerialName("wind_speed_kmh") val windSpeedKmh: Double? = null,
    val advice: String? = null,
    val provider: String? = null,
    val status: String? = null,
    @SerialName("error_reason") val errorReason: String? = null
)

@Serializable
data class DailyForecastDto(
    val date: String,
    @SerialName("temperature_max_c") val temperatureMaxC: Double? = null,
    @SerialName("temperature_min_c") val temperatureMinC: Double? = null,
    @SerialName("apparent_temperature_max_c") val apparentTemperatureMaxC: Double? = null,
    @SerialName("apparent_temperature_min_c") val apparentTemperatureMinC: Double? = null,
    val condition: String? = null,
    @SerialName("condition_code") val conditionCode: Int? = null,
    @SerialName("precipitation_probability_pct") val precipitationProbabilityPct: Int? = null,
    @SerialName("precipitation_sum_mm") val precipitationSumMm: Double? = null,
    val sunrise: String? = null,
    val sunset: String? = null
)

@Serializable
data class WeatherResponseDto(
    @SerialName("location_name") val locationName: String? = null,
    val current: CurrentWeatherDto? = null,
    @SerialName("forecast_daily") val forecastDaily: List<DailyForecastDto> = emptyList()
)

File: mobile/ios/OTravelz/Networking/DTO/WeatherDTOs.swift
import Foundation

public struct CurrentWeatherDTO: Codable, Sendable {
    public let locationName: String?
    public let lat: Double?
    public let lon: Double?
    public let observedAt: String?
    public let temperatureC: Double?
    public let apparentTemperatureC: Double?
    public let condition: String?
    public let conditionCode: Int?
    public let isDay: Int?
    public let humidityPct: Int?
    public let windSpeedKmh: Double?
    public let advice: String?
    public let provider: String?
    public let status: String?

    enum CodingKeys: String, CodingKey {
        case locationName = "location_name"
        case lat
        case lon
        case observedAt = "observed_at"
        case temperatureC = "temperature_c"
        case apparentTemperatureC = "apparent_temperature_c"
        case condition
        case conditionCode = "condition_code"
        case isDay = "is_day"
        case humidityPct = "humidity_pct"
        case windSpeedKmh = "wind_speed_kmh"
        case advice
        case provider
        case status
    }
}

public struct DailyForecastDTO: Codable, Sendable {
    public let date: String
    public let temperatureMaxC: Double?
    public let temperatureMinC: Double?
    public let condition: String?
    public let conditionCode: Int?
    public let precipitationProbabilityPct: Int?

    enum CodingKeys: String, CodingKey {
        case date
        case temperatureMaxC = "temperature_max_c"
        case temperatureMinC = "temperature_min_c"
        case condition
        case conditionCode = "condition_code"
        case precipitationProbabilityPct = "precipitation_probability_pct"
    }
}

public struct WeatherResponseDTO: Codable, Sendable {
    public let locationName: String?
    public let current: CurrentWeatherDTO?
    public let forecastDaily: [DailyForecastDTO]?

    enum CodingKeys: String, CodingKey {
        case locationName = "location_name"
        case current
        case forecastDaily = "forecast_daily"
    }
}
